ssh_keys.go

```diff
@@ -0,0 +1,185 @@
+package paperspace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Paperspace/go-graphql-client"
+)
+
+// SSHKey defines an SSH Key registered with Paperspace.
+type SSHKey struct {
+	ID         string
+	Name       string
+	PublicKey  string
+	DtCreated  time.Time
+	DtModified time.Time
+	DtDeleted  *time.Time
+}
+
+// CreateSSHKey calls the createSSHKey mutation.
+func (c Client) CreateSSHKey(ctx context.Context, x CreateSSHKeyInput) (*SSHKey, error) {
+	m := new(sshKeyCreateMutation)
+	if err := c.graphql.Mutate(ctx, m, x.vars()); err != nil {
+		return nil, fmt.Errorf("createSSHKey: request failed %v", err)
+	}
+	return m.sshKey(x.Name, x.PublicKey), nil
+}
+
+// CreateSSHKeyInput is the arg to CreateSSHKey.
+type CreateSSHKeyInput struct {
+	Name      string `json:"name"`
+	PublicKey string `json:"publicKey"`
+}
+
+func (x CreateSSHKeyInput) vars() map[string]interface{} {
+	return inputvars(x)
+}
+
+type sshKeyCreateMutation struct {
+	CreateSSHKey struct {
+		SSHKey struct {
+			ID         graphql.String
+			DtCreated  graphql.String
+			DtModified graphql.String
+		} `graphql:"sshKey"`
+	} `graphql:"createSSHKey(input: $input)"`
+}
+
+func (m *sshKeyCreateMutation) sshKey(name string, publicKey string) *SSHKey {
+	x := m.CreateSSHKey.SSHKey
+	dtc, _ := time.Parse(time.RFC3339, string(x.DtCreated))
+	dtm, _ := time.Parse(time.RFC3339, string(x.DtModified))
+	return &SSHKey{
+		ID:         string(x.ID),
+		Name:       name,
+		PublicKey:  publicKey,
+		DtCreated:  dtc,
+		DtModified: dtm,
+	}
+}
+
+// DeleteSSHKey calls the deleteSSHKey mutation.
+func (c Client) DeleteSSHKey(ctx context.Context, x DeleteSSHKeyInput) (*SSHKey, error) {
+	m := new(sshKeyDeleteMutation)
+	if err := c.graphql.Mutate(ctx, m, x.vars()); err != nil {
+		return nil, fmt.Errorf("deleteSSHKey: request failed %v", err)
+	}
+	return m.sshKey(x.ID), nil
```

Comment:
i'm new to graphql in go, so sorry if this is a bad question. why does this need a parameter when we've defined the mutation to only have one output?

Reply:
The idea behind this pattern is to define an `sshKey()` factory method off the mutation (declared L81) and since `SSHKey.ID` is part of the input (declared L73), it's not declared as part of the returned gql query. Thus the need to pass it into the factory method.

Reply:
oh. just add that field to what is requested in the mutation
```go
	SSHKey struct {
			ID       graphql.String
			Name       graphql.String
```
i see you left name out of the getSSHKey query also, and then have vars(name), you could just get it from graphql.

Reply:
Maybe that's a better way to go for a more consistent dev experience. Will give it a shot.